### PR TITLE
simulator mavlink don't send_controls() until ready

### PR DESCRIPTION
**Pull request by dagar**
~~- fixes https://github.com/PX4/Firmware/issues/11380~~

### CODE REVIEW

**Comment by dagar** [comment]
@julianoes this is looking promising.

**Comment by dagar** [comment]
Unfortunately still seeing the failure occasionally, although at a much lower rate.

image

**Comment by dagar** [comment]
@julianoes even though this doesn't completely eliminate the problem I think these changes are still worth merging. Please review.

Currently in master (and now in this PR as well) the FW SITL tests are disabled until we have a full solution to https://github.com/PX4/Firmware/issues/11380.

File: src/modules/simulator/simulator.h [hunk 1]
```diff
@@ -315,12 +273,12 @@ class Simulator : public ModuleParams
 	static Simulator *_instance;
 
 	// simulated sensor instances
-	simulator::Report<simulator::RawAccelData>	_accel;
-	simulator::Report<simulator::RawMPUData>	_mpu;
-	simulator::Report<simulator::RawBaroData>	_baro;
-	simulator::Report<simulator::RawMagData>	_mag;
-	simulator::Report<simulator::RawGPSData>	_gps;
-	simulator::Report<simulator::RawAirspeedData> _airspeed;
+	simulator::Report<simulator::RawAccelData>	_accel{1};
```
Comment (julianoes) [inline src/modules/simulator/simulator.h]: For next time it would be nice to separate refactor/cleanup work from functional changes and make a separate commit. That way it's easier to review.

File: src/modules/simulator/simulator_mavlink.cpp [hunk 2]
```diff
@@ -163,24 +166,30 @@ void Simulator::pack_actuator_message(mavlink_hil_actuator_controls_t &msg, unsi
...
 
-		if (_actuator_outputs_sub[i] < 0 || _actuators[i].timestamp == 0) {
-			continue;
-		}
+	if (updated) {
+		actuator_outputs_s actuators{};
+		orb_copy(ORB_ID(actuator_outputs), _actuator_outputs_sub, &actuators);
+
+		if (actuators.timestamp > 0) {
+			const mavlink_hil_actuator_controls_t hil_act_control = pack_actuator_message(actuators);
```
Comment (julianoes) [inline src/modules/simulator/simulator_mavlink.cpp]: So you removed anything but index 0 without even replacing it with a comment or commit message. I think we should document things like that.

File: src/modules/simulator/simulator_mavlink.cpp [hunk 3]
```diff
@@ -735,7 +731,7 @@ void Simulator::pollForMAVLinkMessages(bool publish)
 	struct sched_param param;
 	(void)pthread_attr_getschedparam(&sender_thread_attr, &param);
 
-	param.sched_priority = SCHED_PRIORITY_DEFAULT;
+	param.sched_priority = SCHED_PRIORITY_ACTUATOR_OUTPUTS + 1;
```
Comment (julianoes) [inline src/modules/simulator/simulator_mavlink.cpp]: Why? And why is this part of this pull request?

File: src/modules/simulator/simulator_mavlink.cpp [hunk 4]
```diff
@@ -73,9 +73,12 @@ const unsigned mode_flag_armed = 128;
 const unsigned mode_flag_custom = 1;
 
 using namespace simulator;
+using namespace time_literals;
 
-void Simulator::pack_actuator_message(mavlink_hil_actuator_controls_t &msg, unsigned index)
+mavlink_hil_actuator_controls_t Simulator::pack_actuator_message(const actuator_outputs_s &actuators)
```
Comment (julianoes) [inline src/modules/simulator/simulator_mavlink.cpp]: I think while you're at it the name `pack` is misleading, and it's already used in the mavlink C header terminology.
What about `actuator_controls_from_actuator_outputs()`?

File: src/modules/simulator/simulator_mavlink.cpp (hunk 2 above)
Comment (dagar) [reply src/modules/simulator/simulator_mavlink.cpp]: There's only a single HIL_ACTUATOR_CONTROLS message with no notion of different groups.

Comment (julianoes) [reply src/modules/simulator/simulator_mavlink.cpp]: Ok fair enough.

File: src/modules/simulator/simulator.h (hunk 1 above)
Comment (dagar) [reply src/modules/simulator/simulator.h]: Yes you're right. Pragmatic laziness :smirk: 
https://github.com/PX4/Firmware/pull/11435

File: src/modules/simulator/simulator_mavlink.cpp (hunk 4 above)
Comment (dagar) [reply src/modules/simulator/simulator_mavlink.cpp]: Updated.

File: src/modules/simulator/simulator_mavlink.cpp (hunk 3 above)
Comment (dagar) [reply src/modules/simulator/simulator_mavlink.cpp]: Comment added.

File: src/modules/simulator/simulator.h (hunk 1 above)
Comment (julianoes) [reply src/modules/simulator/simulator.h]: :smile:

File: src/modules/simulator/simulator_mavlink.cpp (hunk 3 above)
Comment (julianoes) [reply src/modules/simulator/simulator_mavlink.cpp]: Btw. as far as I know these priorities are ignored unless you run px4 as root.
